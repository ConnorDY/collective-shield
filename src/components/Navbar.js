import React, { Component } from 'react';
import { Link } from "react-router-dom";

import navlogo from '../img/navlogo.png'

class Navbar extends Component {

    render() {

        return (
            <nav className="c-navbar">
                <div className="c-navbar__wrapper">
                    <div className="c-navbar__logo">
                        <img src={navlogo} alt="Catamaran Logo" />
                    </div>
                    <div className="c-navbar__items">
                        <ul>
                            <li>
                                <Link to='/' className={this.props.activePage === 'home' ? '-active' : ''}>
                                    Home
                                </Link>
                            </li>
                            <li>
                                <Link to='/makers' className={this.props.activePage === 'makers' ? '-active' : ''}>
                                    Makers
                                </Link>
                            </li>
                            <li>
                                <Link to='/requests' className={this.props.activePage === 'requests' ? '-active' : ''}>
                                    Requests
                                </Link>
                            </li>
                        </ul>
                    </div>
                </div>
            </nav>
        );
    }
}

export default Navbar;
