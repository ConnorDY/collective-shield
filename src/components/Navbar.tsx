import React from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { NavLink, Link } from "react-router-dom";
import { get } from 'lodash';

import User from '../models/User';
import Avatar from './Avatar';
import navLogo from '../assets/img/navlogo.png';

const Navbar: React.FC<{ user: User | undefined }> = ({ user }) => {
  const showAdmin = get(user, 'isSuperAdmin', false);
  return (
    <nav className="nav">
      <Container>
        <Row>
          <Col className="branding">
            <Link to="/">
              <>
                <img alt="Logo" className="logo" src={navLogo}></img>
                <div>MaskShield</div>
              </>
            </Link>
          </Col>

          {
            user &&
            <Col xs={6}>
              <Row className="justify-content-between">
                <Col className="my-auto font-weight-bold text-center">
                  <NavLink to="/" exact activeClassName="active" className="nav-link">Request Shields</NavLink>
                  <NavLink to="/work" activeClassName="active" className="nav-link">Print Shields</NavLink>
                {
                  showAdmin &&
                    <NavLink to="/requests" activeClassName="active" className="nav-link">Admin</NavLink>
                }
                </Col>
              </Row>
            </Col>
          }

          {user ? (
            <Col className="user">
              <Row className="justify-content-end">
                <Col xs={4} className="my-auto font-weight-bold">
                  {user.firstName}
                </Col>
                <Avatar size="40" user={user} />
              </Row>
            </Col>
          ) : (
            <></>
          )}
        </Row>
      </Container>
    </nav>
  );
};

export default Navbar;
