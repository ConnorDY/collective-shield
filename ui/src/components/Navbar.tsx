import React from 'react';
import { Col, Container, Row, Navbar, Nav } from 'react-bootstrap';
import { NavLink, Link } from 'react-router-dom';
import { get } from 'lodash';

import User from '../models/User';
import Avatar from './Avatar';
import Logo from './Logo';

const MainNav: React.FC<{ user: User | undefined }> = ({ user }) => {
  const showAdmin = get(user, 'isSuperAdmin', false);
  const logoFill = ['default', 'blue-purple', 'orange', 'gray'];

  return (
    <Navbar expand="lg" className="nav">
      <Container>
        <Row className="justify-content-end nav-inner no-gutters">
          <Col xs={12} lg={3}>
            <Navbar.Brand href="/" className="branding">
              <Logo height="30" width="30" fill={logoFill[1]} />
              <span
                className={`align-middle logo-text logo-text-${logoFill[0]}`}
              >
                Collective Shield
              </span>
            </Navbar.Brand>
            <Navbar.Toggle aria-controls="main-nav-bar" />
          </Col>
          <Col xs={12} lg={9}>
            <Navbar.Collapse
              id="main-nav-bar"
              className="justify-content-between"
            >
              <Row className="collapse-row no-gutters">
                <Col xs={12} lg={6} className="nav-links">
                  {user && (
                    <Nav className="mr-auto text-lg-center">
                      {showAdmin && (
                        <NavLink
                          to="/requests"
                          activeClassName="active"
                          className="nav-link"
                        >
                          Admin
                        </NavLink>
                      )}
                    </Nav>
                  )}
                </Col>
                <Col xs={12} lg={6} className="user text-lg-right">
                  {user ? (
                    <Navbar.Text className="my-auto  font-weight-bold">
                      <span className="ml-lg-3 mr-3 my-auto">
                        {user.firstName}
                      </span>
                      <Avatar size="40" user={user} />
                      <Link to="/logout" className="ml-lg-3 my-lg-auto">
                        Logout
                      </Link>
                    </Navbar.Text>
                  ) : (
                    <></>
                  )}
                </Col>
              </Row>
            </Navbar.Collapse>
          </Col>
        </Row>
      </Container>
    </Navbar>
  );
};

export default MainNav;
