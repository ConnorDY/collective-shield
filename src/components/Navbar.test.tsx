import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar';

describe('Navbar', () => {
  const props = {
    user: {
      firstName: 'Mary'
    }
  };

  it('is defined', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toBeDefined();
  });

  it('is defined when there is a user object', () => {
    const wrapper = shallow(<Navbar {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when there is a user object', () => {
    const wrapper = shallow(<Navbar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
