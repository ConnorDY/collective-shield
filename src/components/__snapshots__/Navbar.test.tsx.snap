// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Navbar matches snapshot when there is a user object 1`] = `ShallowWrapper {}`;
